pig/src/main/java/com/netflix/iceberg/pig/IcebergStorage.java

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.iceberg.pig;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.netflix.iceberg.Schema;
+import com.netflix.iceberg.Table;
+import com.netflix.iceberg.types.Types;
+import org.apache.hadoop.fs.Path;
+import org.apache.pig.impl.util.ObjectSerializer;
+import org.apache.pig.impl.util.UDFContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.netflix.iceberg.Tables;
+import com.netflix.iceberg.hadoop.HadoopTables;
+import com.netflix.iceberg.pig.IcebergInputFormat.IcebergRecordReader;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.RecordReader;
+
+import org.apache.hadoop.util.ReflectionUtils;
+import org.apache.pig.Expression;
+import org.apache.pig.Expression.*;
+import org.apache.pig.LoadFunc;
+import org.apache.pig.LoadMetadata;
+import org.apache.pig.LoadPredicatePushdown;
+import org.apache.pig.LoadPushDown;
+import org.apache.pig.ResourceSchema;
+import org.apache.pig.ResourceStatistics;
+import org.apache.pig.backend.hadoop.executionengine.mapReduceLayer.PigSplit;
+import org.apache.pig.data.Tuple;
+import org.apache.pig.impl.logicalLayer.FrontendException;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static com.netflix.iceberg.expressions.Expressions.*;
+import static org.apache.pig.Expression.OpType.*;
+import static com.netflix.iceberg.pig.IcebergInputFormat.*;
```

Comment:
Nit: I prefer not using wildcard imports so that it is easy to see where symbols are defined. For example, this pulls in `ICEBERG_FILTER_EXPRESSION` used below, but it's not obvious where that is defined if you're looking at the code in github and not an editor.